processor/tailsamplingprocessor/internal/sampling/and.go

```diff
@@ -45,7 +45,7 @@ func (c *And) Evaluate(traceID pcommon.TraceID, trace *TraceData) (Decision, err
 		if err != nil {
 			return Unspecified, err
 		}
-		if decision == NotSampled {
+		if decision == NotSampled || decision == InvertNotSampled {
```

Comment:
Nice find, now I'm wondering if [other places which evaluate decision](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/tailsamplingprocessor/internal/sampling/composite.go#L112) should be updated as well. Quickly looking at the code, this seems to be the case (though to be fair I haven't looked at tailsamplingprocessor code in a while and might be missing something)

Reply:
I admit I haven't dived deep into the string matching, but it did make me wonder why it matters if it's `Sampled` or `InvertSampled`. It wasn't immediately obvious why a regular decision and an invert decision would need to be treated differently.

Reply:
It seems it's the case as well, I tested it and it failed. I am opening second PR right now :)

Reply:
@tillig I don't fully understand how the invert works but it looks like there's something related to the `InvertSampled` decision in the [makeDecision](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/fe12a901958f7ed8dfce7f14800776ca5168c37e/processor/tailsamplingprocessor/processor.go#L209) function

Reply:
Yeah, but even that seems really sort of weird to me.

```golang
	// InvertNotSampled takes precedence over any other decision
	if samplingDecision[sampling.InvertNotSampled] {
		finalDecision = sampling.NotSampled
	} else if samplingDecision[sampling.Sampled] {
		finalDecision = sampling.Sampled
	} else if samplingDecision[sampling.InvertSampled] && !samplingDecision[sampling.NotSampled] {
		finalDecision = sampling.Sampled
	}
```

- All the policies get executed before a decision is made. That's weird, and _sounds_ like it's more of a composite thing than it should be. It _feels_ like the policies should be executed in order and the first one that says "sample this thing" would allow sampling through. That'd perform better, too.
- `InvertNotSampled` has the _highest_ precedence, but `InvertSampled` has the _lowest_ precedence and has to be specifically checked to see if it's also not contradicted by `NotSampled`.

I dunno, I'm coming at it late, I just don't see any doc on it and the special treatment there is kind of confusing. It gets even more confusing when I think about how this affects the way I might get an unexpected result from just a series of policies now. If I put an invert match policy in the mix and it says "no," it'll countermand any other policy I put in there. Even if it's not in an `and` policy. 🤔  Am I reading that wrong?

Reply:
Sure thing, I was curious if you would find that useful. As for trying it out, it should be quite easy to [include it via a builder](https://github.com/SumoLogic/sumologic-otel-collector/blob/main/otelcolbuilder/.otelcol-builder.yaml#L36-L37)

Reply:
Cool, thanks. If/when that time comes, I'll have to actually start learning what "include it via a builder" means. I'm not a go dev and there's a long tail of things to learn. 😄

Reply:
Yeah, I agree with that, the behavior of the `tailsamplingprocessor` is not what i would expect. 
the readme should definitely be updated to reflect that.
I know there was talk about deprecating the `tailsamplingprocessor` so I like the idea of upstreaming [cascadingfilterprocessor](https://github.com/SumoLogic/sumologic-otel-collector/tree/main/pkg/processor/cascadingfilterprocessor) I think it can be a great fit to replace it.